spec/lib/dfe_sign_in/api/user_spec.rb

```diff
@@ -1,72 +1,112 @@
 
-  context "with a valid response" do
-    let(:status) { 200 }
-    let(:response) do
-      {
-        "userId" => "999",
-        "serviceId" => "123",
-        "organisationId" => "456",
-        "roles" => [
-          {
-            "id" => "role-id",
-            "name" => "My role",
-            "code" => "my_role",
-            "numericId" => "9999",
-            "status" => {
-              "id" => 1,
-            },
-          },
-        ],
-        "identifiers" => [
-          {
-            "key" => "identifier-key",
-            "value" => "identifier-value",
-          },
-        ],
-      }
+        expect(users.second.given_name).to eq("Bob")
+
+        expect(users.third.given_name).to eq("Eve")
+      end
     end
 
-    describe "has_role?" do
-      let(:has_role?) { subject.has_role?(role) }
+    context "with multiple pages" do
+      let!(:first_page_stub) { stub_dfe_sign_in_user_list_request(number_of_pages: 3) }
+      let!(:second_page_stub) { stub_dfe_sign_in_user_list_request(number_of_pages: 3, page_number: 2) }
+      let!(:third_page_stub) { stub_dfe_sign_in_user_list_request(number_of_pages: 3, page_number: 3) }
 
-      context "when a role exists" do
-        let(:role) { "my_role" }
-        it { expect(has_role?).to eq(true) }
-      end
+      it "requests multiple pages" do
+        expect(users.count).to eq(9)
 
-      context "when a role does not exist" do
-        let(:role) { "other_role" }
-        it { expect(has_role?).to eq(false) }
+        expect(first_page_stub).to have_been_requested
+        expect(second_page_stub).to have_been_requested
+        expect(third_page_stub).to have_been_requested
       end
     end
+  end
+
+  describe "#new" do
+    subject { described_class.new(user_id: 999, organisation_id: 456) }
+
+    let(:url) { "#{DfeSignIn.configuration.base_url}/services/#{DfeSignIn.configuration.client_id}/organisations/456/users/999" }
+
+    before do
+      stub_request(:get, url)
+        .to_return(body: response.to_json, status: status)
+    end
+
+    context "with a valid response" do
+      let(:status) { 200 }
+      let(:response) do
+        {
+          "userId" => "999",
+          "serviceId" => "123",
+          "organisationId" => "456",
+          "roles" => [
+            {
+              "id" => "role-id",
+              "name" => "My role",
+              "code" => "my_role",
+              "numericId" => "9999",
+              "status" => {
+                "id" => 1,
+              },
+            },
+          ],
+          "identifiers" => [
+            {
+              "key" => "identifier-key",
+              "value" => "identifier-value",
+            },
+          ],
+        }
+      end
 
-    describe "#role_codes" do
-      it "returns the role codes" do
-        expect(subject.role_codes).to eq(["my_role"])
+      describe "has_role?" do
+        let(:has_role?) { subject.has_role?(role) }
+
+        context "when a role exists" do
```

Comment:
I am battliing with how I feel about conexts like this at the moment, I suspect some people like them and it reflects their mental model, whilst other people, the added processing and nesting burdens them, I do wonder if in a simple case like this they bring complexity where it may not be needed, how would something like 

````ruby
describe "#has_role?" do
  it "returns true when the user has the role" do
    expect(subject.has_role?("my_role")).to eq(true)
    expect(subject.has_role?("other_role")).to eq(false)
  end
end
````
feel to you?

Reply:
👍